src/test/scala-3/com/raquo/airstream/split/SplittableTypeSpec.scala

```diff
@@ -0,0 +1,474 @@
+package com.raquo.airstream.split
+
+import com.raquo.airstream.UnitSpec
+import com.raquo.airstream.eventbus.EventBus
+import com.raquo.airstream.fixtures.{Effect, TestableOwner}
+import com.raquo.airstream.state.Var
+import com.raquo.airstream.split.SplittableTypeMacros.*
+
+import scala.collection.{immutable, mutable}
+import scala.scalajs.js
+import com.raquo.airstream.ShouldSyntax.shouldBeEmpty
+
+class SplittableTypeSpec extends UnitSpec {
+
+  sealed trait Foo
+
+  final case class Bar(strOpt: Option[String]) extends Foo
+  enum Baz extends Foo {
+    case Baz1, Baz2
+  }
+  case object Tar extends Foo
+
+  final case class Res(result: Any)
+  
+  it("split match signal") {
+    val effects = mutable.Buffer[Effect[String]]()
+
+    val myVar = Var[Foo](Bar(Some("initial")))
+
+    val owner = new TestableOwner
+
+    val signal = myVar.signal
+      .splitMatch
+      .handleCase {
+        case Bar(Some(str)) => str
+        case Bar(None) => "null"
+      } { case (str, strSignal) =>
```

Comment:
I just noticed that the public API requires the user to provide a function expecting a single argument – a tuple of `(A, Signal[A])` – does it have to be like this, instead of accepting two arguments, `A` and `Signal[A]`?

The single tuple argument basically requires the end user to provide a partial function using `case`, where a total function is expected. If the function accepted two arguments, we could write the same but without the second `case`:

```
      .handleCase {
        case Bar(Some(str)) => str
        case Bar(None) => "null"
      } { (str, strSignal) =>
        ...
      }
```

Reply:
It can be changed, it's just an old code from initial implementations that slip through